lib/src/components/NxFilterInput/__tests__/NxFilterInput.test.tsx

```diff
@@ -96,6 +99,38 @@ describe('NxFilterInput', function() {
     expect(icon).toHaveClassName('nx-icon--filter-icons');
   });
 
+  it('passes an NxButton with the Close icon and nx-btn--clear classname as the suffixContent', function() {
+    const BtnFixture = function() {
+          return shallowComponent().prop('suffixContent');
+        },
+        btn = shallow(<BtnFixture />);
+
+    expect(btn).toExist();
+    expect(btn).toMatchSelector(NxButton);
+    expect(btn.children()).toMatchSelector(Close);
+    expect(btn).toHaveClassName('nx-btn--clear');
+  });
+
+  it('clears input text when the Escape key is pressed', function() {
+    const onKeyDown = jest.fn(),
+        component = shallowComponent({ onKeyDown, value: 'a' });
+
+    component.simulate('keyDown', { key: 'Escape' });
+
+    expect(component).toHaveText('');
+  });
+
+  it('clears input text when the clear filter button is clicked', function() {
```

Comment:
This is a stateless component.  The effect of the button/ESC shouldn't really be to clear the text, it should be to fire `onChange` with the empty string.  It's up to typical calling code to respond to that by passing `''` as the new `value`, causing the component text to actually become empty.

Reply:
Update here cc5699c70171de5574aae0a2e70aa4bbe6cad3b5.
The RTL unit tests will be in another PR.